### PR TITLE
fix(wallet): fix currency selector in wallet panel

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/24357

## Submitter Checklist:

- [x] I confirm that no security/privacy review is not needed
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues/24357) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Open wallet panel
2. Click on buy tab
3. In the amount input, click on the currency symbol, a `Select currency` view should open.
4. Select a currency from the list displayed, the currency selection view will hide and the currency symbol will change to the symbol of the currency you have selected.

https://user-images.githubusercontent.com/48117473/182351905-ea572ffe-fcff-4624-8623-139cf0af3bc3.mov

### CODE REVIEW

**Review by Soosay86 — COMMENTED** [review]
I need more details about this brave account.

**Comment by muliswilliam** [comment]
> I need more details about this brave account.

Which brave account?

File: components/brave_wallet_ui/panel/container.tsx [hunk 1]
```diff
@@ -889,6 +903,27 @@ function Container () {
...
+        <StyledExtensionWrapper>
+          <Panel
+            navAction={navigateTo}
+            title={panelTitle}
+            useSearch={false}
+          >
+            <ScrollContainer>
+              <SelectCurrency
+                onBack={onBack}
+                onSelectCurrency={onSelectCurrency}
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/panel/container.tsx:918]: Could we utilize the `navigateTo` method here for these 2 props? 🙂

`onBack={() => navigateTo('buy')`
`onSelectCurrency={() => navigateTo('buy')`

File: components/brave_wallet_ui/panel/container.tsx [hunk 2]
```diff
@@ -758,6 +771,7 @@ function Container () {
               <Buy
                 onChangeBuyView={onChangeSendView}
                 selectedAsset={selectedBuyAsset}
+                onShowCurrencySelection={onShowCurrencySelection}
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/panel/container.tsx:774]: Could do the same here?

File: components/brave_wallet_ui/panel/container.tsx (hunk 1 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/panel/container.tsx:918]: Fixed here: https://github.com/brave/brave-core/commit/c2a4f8d2b7834199a805d767fc829c80cb904aee

File: components/brave_wallet_ui/panel/container.tsx (hunk 2 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/panel/container.tsx:774]: @Douglashdaniel The function is already using `navigateTo` function internally as show here: https://github.com/brave/brave-core/blob/c2a4f8d2b7834199a805d767fc829c80cb904aee/components/brave_wallet_ui/panel/container.tsx#L426-L428

**Review by Douglashdaniel — APPROVED** [review]
++

**Pull request merged**